pull_buildpack.go

```diff
@@ -0,0 +1,80 @@
+package pack
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/buildpacks/pack/config"
+	"github.com/buildpacks/pack/internal/buildpack"
+	"github.com/buildpacks/pack/internal/buildpackage"
+	"github.com/buildpacks/pack/internal/dist"
+	"github.com/buildpacks/pack/internal/style"
+)
+
+type PullBuildpackOptions struct {
+	URI          string
+	RegistryType string
+	RegistryURL  string
+	RegistryName string
+}
+
+func (c *Client) PullBuildpack(ctx context.Context, opts PullBuildpackOptions) error {
+	locatorType, err := buildpack.GetLocatorType(opts.URI, []dist.BuildpackInfo{})
+	if err != nil {
+		return err
+	}
+
+	switch locatorType {
+	case buildpack.URILocator:
+		err := ensureBPSupport(opts.URI)
+		if err != nil {
+			return errors.Wrapf(err, "checking support")
+		}
+
+		blob, err := c.downloader.Download(ctx, opts.URI)
+		if err != nil {
+			return errors.Wrapf(err, "downloading buildpack from %s", style.Symbol(opts.URI))
+		}
+
+		isOCILayout, err := buildpackage.IsOCILayoutBlob(blob)
+		if err != nil {
+			return errors.Wrapf(err, "checking format")
+		}
+
+		if isOCILayout {
+			_, _, err = buildpackage.BuildpacksFromOCILayoutBlob(blob)
+			if err != nil {
+				return errors.Wrapf(err, "extracting buildpacks from %s", style.Symbol(opts.URI))
+			}
+		} else {
+			return errors.Errorf("buildpack is in an unsupported layout")
+		}
+	case buildpack.PackageLocator:
```

Comment:
Can you add a test for this? I see a test for a `URI` and a registry BP, but none for a PackageLocator (unless I missed it)

Reply:
The "pulling from a docker registry" test is for `PackageLocator`. I've removed the `URILocator` code path.